loadtest/control/simulcontroller/controller.go

```diff
@@ -48,16 +49,123 @@ func (c *SimulController) Run() {
 
 	defer c.sendStopStatus()
 
+	if resp := control.SignUp(c.user); resp.Err != nil {
+		c.status <- c.newErrorStatus(resp.Err)
+	} else {
+		c.status <- c.newInfoStatus(resp.Info)
+	}
+
+	if resp := control.Login(c.user); resp.Err != nil {
+		c.status <- c.newErrorStatus(resp.Err)
+	} else {
+		c.status <- c.newInfoStatus(resp.Info)
+		c.connect()
+	}
+
+	if resp := c.reload(false); resp.Err != nil {
+		c.status <- c.newErrorStatus(resp.Err)
+	} else {
+		c.status <- c.newInfoStatus(resp.Info)
+	}
+
+	if resp := control.JoinTeam(c.user); resp.Err != nil {
+		c.status <- c.newErrorStatus(resp.Err)
+	} else {
+		c.status <- c.newInfoStatus(resp.Info)
+	}
+
+	actions := []userAction{
+		{
+			run: func(u user.User) control.UserActionResponse {
+				return c.reload(false)
+			},
+			frequency: 1,
+		},
+		{
+			run:       control.JoinTeam,
+			frequency: 1,
+		},
+		{
+			run:       control.JoinChannel,
+			frequency: 1,
+		},
+		{
+			run:       control.SearchPosts,
+			frequency: 1,
+		},
+		{
+			run:       control.SearchChannels,
+			frequency: 1,
+		},
+		{
+			run:       control.SearchUsers,
+			frequency: 1,
+		},
+		{
+			run:       control.ViewUser,
+			frequency: 1,
+		},
+		{
+			run:       control.CreatePost,
+			frequency: 1,
+		},
+		{
+			run:       control.AddReaction,
+			frequency: 1,
+		},
+		{
+			run:       control.UpdateProfileImage,
+			frequency: 1,
+		},
+		{
+			run:       control.CreateGroupChannel,
+			frequency: 1,
+		},
+		{
+			run:       control.CreateDirectChannel,
+			frequency: 1,
+		},
+		{
+			run:       control.ViewChannel,
+			frequency: 1,
+		},
+		{
+			run:       control.LeaveChannel,
+			frequency: 1,
+		},
+		{
+			run:       control.RemoveReaction,
+			frequency: 1,
+		},
+	}
+
 	for {
-		// TODO: implement picking an action and run it.
+		action := pickAction(actions)
```

Comment:
Action may return nil, it should be handled.

Reply:
True but that's never going to happen unless the slice of actions is empty and that's hardcoded not to be empty.
At some point if we start passing the actions from a config file we'll parse it and fail before getting here.
We could check for nil and panic. What we don't want is the controller to keep running with no actions. 
I think we can guarantee action will never be nil at this point.

Reply:
Hmm, you may be right on guaranteeing that actions never going to be empty here. But I would prefer using `action, err` return on `pickAction()` function.

Reply:
We can add an error, just saying we'll then be panicking cause at that point it's not recoverable from the controller's perspective.